internal/apiserver/backrestservice/backrestimpl.go

```diff
@@ -279,7 +280,13 @@ func getPrimaryPodName(cluster *crv1.Pgcluster, ns string) (string, error) {
 
 	//look up the backrest-repo pod name
 	selector := "pg-cluster=" + cluster.Spec.Name + ",pgo-backrest-repo=true"
-	repopods, err := apiserver.Clientset.CoreV1().Pods(ns).List(metav1.ListOptions{LabelSelector: selector})
+
+	options := metav1.ListOptions{
+		FieldSelector: fields.OneTermEqualSelector("status.phase", string(v1.PodRunning)).String(),
+		LabelSelector: selector,
```

Comment:
Why not add this to the options struct too? You would have to redefine `selector` somewhere else but could be cleaner
```suggestion
		LabelSelector: "pg-cluster=" + cluster.Spec.Name + ",pgo-backrest-repo=true",
```

Reply:
The main goal was to add the field selector and not make extensive changes to the code, so the primary issue could be fixed.

If I wer eto clean up the selector, I would follow a similar pattern to how we're constructing `FieldSelector`.